File: frontend/setup_env.js
#!/usr/bin/env node
/**
 * Script para configurar el entorno de desarrollo del frontend.
 * Este script ayuda a configurar las variables de entorno necesarias.
 */

const fs = require('fs');
const path = require('path');

function setupEnvironment() {
    console.log('🚀 Configurando entorno de desarrollo del frontend...');
    
    // Directorio actual
    const currentDir = __dirname;
    
    // Archivos de entorno
    const envDevelopment = path.join(currentDir, '.env.development');
    const envProduction = path.join(currentDir, '.env.production');
    const envFile = path.join(currentDir, '.env');
    
    if (!fs.existsSync(envFile)) {
        console.log('📝 Creando archivo .env desde .env.development...');
        fs.copyFileSync(envDevelopment, envFile);
        console.log('✅ Archivo .env creado exitosamente');
    } else {
        console.log('⚠️ El archivo .env ya existe');
    }
    
    console.log('\n📋 Variables de entorno configuradas:');
    console.log('   • REACT_APP_API_URL: URL de la API backend');
    console.log('   • REACT_APP_ENVIRONMENT: Entorno de ejecución');
    console.log('   • REACT_APP_DEBUG: Modo debug activado/desactivado');
    
    console.log('\n⚠️ IMPORTANTE:');
    console.log('   • Las variables deben empezar con REACT_APP_ para ser accesibles');
    console.log('   • Configura la URL correcta de tu API backend');
    console.log('   • No subas archivos .env al repositorio');
    
    console.log('\n🎉 Configuración completada!');
}

if (require.main === module) {
    setupEnvironment();
}

module.exports = { setupEnvironment };

File: backend/setup_env.py
#!/usr/bin/env python3
"""
Script para configurar el entorno de desarrollo del backend.
Este script ayuda a configurar las variables de entorno necesarias.
"""

import os
import shutil
from pathlib import Path

def setup_environment():
    """Configura el entorno de desarrollo."""
    print("🚀 Configurando entorno de desarrollo...")
    
    # Directorio actual
    current_dir = Path(__file__).parent
    
    # Archivos de entorno
    env_development = current_dir / ".env.development"
    env_production = current_dir / ".env.production"
    env_file = current_dir / ".env"
    
    if not env_file.exists():
        print("📝 Creando archivo .env desde .env.development...")
        shutil.copy(env_development, env_file)
        print("✅ Archivo .env creado exitosamente")
    else:
        print("⚠️ El archivo .env ya existe")
        
    # Crear directorio de uploads si no existe
    uploads_dir = current_dir / "uploads"
    uploads_dir.mkdir(exist_ok=True)
    print("📁 Directorio de uploads configurado")
    
    print("\n📋 Variables de entorno configuradas:")
    print("   • DATABASE_URL: Conexión a PostgreSQL")
    print("   • SECRET_KEY: Clave secreta para JWT")
    print("   • UPLOAD_DIR: Directorio para archivos subidos")
    print("   • ALLOWED_ORIGINS: Orígenes permitidos para CORS")
    
    print("\n⚠️ IMPORTANTE:")
    print("   • Asegúrate de cambiar el SECRET_KEY en producción")
    print("   • Configura tu base de datos PostgreSQL")
    print("   • No subas el archivo .env al repositorio")
    
    print("\n🎉 Configuración completada!")

if __name__ == "__main__":
    setup_environment()
